### ISSUE TITLE
[JENKINS-19838] Ability to use separate polling queues per SCM type

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#3646]
<p>Some SCMs are quite fast at pollings while others are very slow (Eg. Git is fast with low load on the server, while CVS is slow with high load on the server).</p>

<p>It would be nice to have separate polling queues per SCM type, so the fast Git pollings are not blocked by the slow ones.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/saschavogt.jpg" title="saschavogt's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=saschavogt">saschavogt</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-19838" target="_blank">Ability to use separate polling queues per SCM type</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Some SCMs are quite fast at pollings while others are very slow (Eg. Git is fast with low load on the server, while CVS is slow with high load on the server).</p>

<p>It would be nice to have separate polling queues per SCM type, so the fast Git pollings are not blocked by the slow ones.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>all</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-19838?focusedId=188494&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-188494">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Pull request had a lively conversation about how to go about this.</p>
</pre>
</details>

</ul>
</details>

<p>Pull request had a lively conversation about how to go about this.</p>
